tools/scripts/pre-release.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -ex
+
+## Pre release hook. Run from the root of the project. Any arguments passed in are directly
+## sent to `bumpversion`
+
+PROJECT_ROOT=$(pwd)
+
+if [[ $(git diff --shortstat 2> /dev/null | tail -n1) != "" ]]; then
+    echo "git: there are dirty files."
+    exit 1
+elif [[ $(git status --porcelain 2>/dev/null| grep "^??") ]]; then
+    echo "git: there are untracked files."
+    exit 1
+fi
+
+## lock in current protobuf state
+# Buf image binary
+BUF_IMAGE=buf.image.bin
+cd proto
+make gen-buf-image
+git add $BUF_IMAGE
+git commit -m "lock backward buf compatibility check" || echo "buf image is already up to date"
+cd $PROJECT_ROOT
+
+## bump version
+bumpversion $@
```

Comment:
blocking: upon consideration, I think it's actually better to not do this for the user, and have two separate steps that are manual (and documented in the release process).

(this also means we should change the script name)

Reply:
this makes sense to me and sounds like a smaller change as well.

Reply:
done. any suggestions on the name?